### ISSUE TITLE
Unhandled Debug Exception rised in file 'tuple.hpp'.

### ISSUE THREAD

**Issue by ChkVienna** [rpclib/rpclib#264]
I'm starting with rpclib and doing the Examples in Guides > Primer. Great thanks for this library, you do a great job with all your contributors.
I could build rpclib on Win10 with VS2015-prof and in VS2019-enterprise, however in Win32 configuration only. In x64 I could build rpclib, but not my tutorial projects(?).

I did the first two examples in 'Primer' about server and client with the proposed error handling. And here I got the 'Unhandled exception error' in the client in Debug-Configuration. In Release-Configuration the client stoped execution at the same location without any Message. See the verbose Comment at the end of main() in the Client-Source below.

//rpc-client_1
/*
    Example copied from page:
    http://rpclib.net/primer/
*/
#include <iostream>
#include "rpc/client.h"
#include "rpc/rpc_error.h"

int main() {
    rpc::client c("localhost", 8080);
    try {
        std::cout << "add(2, 3) = ";
        double five = c.call("add", 2, 3).as<double>();
        std::cout << five << std::endl;

        std::cout << "sub(3, 2) = ";
        double one = c.call("sub", 3, 2).as<double>();
        std::cout << one << std::endl;

        std::cout << "mul(5, 0) = ";
        double zero = c.call("mul", five, 0).as<double>();
        std::cout << zero << std::endl;

        std::cout << "div(3, 0) = ";
        double hmm = c.call("div", 3, 0).as<double>();
        std::cout << hmm << std::endl;
    }
    catch (rpc::rpc_error &e) {
        std::cout << std::endl << e.what() << std::endl;
        std::cout << "in function " << e.get_function_name() << ": ";

        //return 1; // because the following code has a Runtime-Error

        using err_t = std::tuple<int, std::string>;
        //The next line produces a Runtime-Error:
        //Unhandled exception at 0x76E946D2 in rpc-client_1.exe:
        //        Microsoft C++ exception: clmdep_msgpack::v1::type_error at memory location 0x005DE660.
        //The Debugger shows line 126 'StdTupleConverter<decltype(v), sizeof...(Args)>::convert(o, v);'
        //        in file 'tuple.hpp'.
        auto err = e.get_error().as<err_t>();
        std::cout << "[error " << std::get<0>(err) << "]: " << std::get<1>(err)
            << std::endl;
        return 1;
    }  // */
}

[rpc-client_1.cpp.txt](https://github.com/rpclib/rpclib/files/5903162/rpc-client_1.cpp.txt)
[rpc-server_1.cpp.txt](https://github.com/rpclib/rpclib/files/5903164/rpc-server_1.cpp.txt)

**Comment by ChkVienna** [comment]
The question is still open. Closing was because of my beginnerism to GitHub :-)

**Comment by ChkVienna** [comment]
I have found the reason: It is caused by the server.
This is the incorrect statement:
    rpc::this_handler().respond_error("Division by zero");
and is one of the examples given on http://rpclib.net Guides > Primer, chapter 'Responding With Errors'.

The correct statement is:
    rpc::this_handler().respond_error(
        std::make_tuple(1, "Division by zero"));
I found it in the provided example-code of 
examples > calculator > calculator_server.cc
and it's also written in Guides > Primer one paragraph later.
